src/bsp/include/bsp/ISpiStm.h

```diff
@@ -1,29 +1,43 @@
 #ifndef __ISPISTM_H__
 #define __ISPISTM_H__
 
+#include "common/IProtobufStream.h"
 #include <cstdint>
 
 #define CRC32_SIZE (sizeof(uint32_t))
 #define STM_SPI_MAX_MESSAGE_LENGTH (2048u - CRC32_SIZE)
 
-class ISpiStm {
+class ISpiStm : public IProtobufStream {
   public:
-    virtual ~ISpiStm() = default;
+    ~ISpiStm() override = default;
 
     /**
      * @brief Sends a buffer to the Stm via spi by appending with the length and the CRC32
-     * (non-blocking and thread-safe)
      * @param buffer Pointer to the data to send
      * @param length Number of bytes to send
      * @return True if transfer started. False otherwise.
      */
-    virtual bool send(const uint8_t* buffer, uint16_t length) = 0;
+    bool send(const uint8_t* buffer, uint16_t length) override = 0;
+
+    /**
+     * @brief Receive data from spi inside a buffer
+     * @param data The buffer in which to store the data received
+     * @param length The number of bytes to read
+     * @return true if successful, false otherwise
+     */
+    bool receive(uint8_t* data, uint16_t length) override = 0;
```

Comment:
Is there a reason why you removed the virtual and used the override? You could make it virtual instead of overriding them since you make them purely virtual, it's a bit more verbose on the intent and a bit strange to override with a purely virtual method. Same for send and the destructor.

Just so you know are not required to write the virtual (or even the whole function definition since you inherit them) but it's makes it easier to read the .h.